flocker/node/_deploy.py

```diff
@@ -135,6 +129,8 @@ def __init__(self, node_uuid, local_applications, leases):
         :param applications: Applications running on the node.
         :param Leases leases: The current leases on datasets.
         """
+        if local_applications is None:
+            local_applications = []
```

Comment:
I don't entirely know. This was one of the functional changes in Itamar's
original sketch.

I believe it works because an absent container agent will mean
this function gets called with None. This then treats that as []
so it carries on as if there are no containers.